docs/compass/03-automatic-scenario-assignment.md

```diff
@@ -0,0 +1,213 @@
+
+
+Director API contains the following mutations for managing Automatic Scenario Assignments.
+```graphql
+   createAutomaticScenarioAssignment(in: AutomaticScenarioAssignmentSetInput!): AutomaticScenarioAssignment 
+   deleteAutomaticScenarioAssignmentForScenario(scenarioName: String!): AutomaticScenarioAssignment 
+   deleteAutomaticScenarioAssignmentsForSelector(selector: LabelSelectorInput!): [AutomaticScenarioAssignment!]! 
+}
+```
+When creating Assignment, the following conditions are checked:
+* for given scenario, at most one Assignment exists
+* given scenario has to exists
+* selector's value type is a string
+
+Director API contains queries for fetching all assignments, assignments for the given Scenario and the given label selector.
+```graphql
+   automaticScenarioAssignments(first: Int = 100, after: PageCursor): AutomaticScenarioAssignmentPage 
+   automaticScenarioAssignmentForScenario(scenarioName: String!): AutomaticScenarioAssignment 
+   automaticScenarioAssignmentsForSelector(selector: LabelSelectorInput!): [AutomaticScenarioAssignment!]! 
+```
+
+## Example
+Let assume a situation, that Integration System that is responsible for registering Runtimes, label every runtime with information about a user who triggered provisioning a Runtime. 
+Then, you can define an Assignment, that every Runtime provisioned by a given person belongs to the specific Scenario.
+
+1. Create or update `scenarios` Label Definition and specify the following scenarios: 
+* DEFAULT
+* MARKETING
+* WAREHOUSE
+
+```graphql
+mutation  {
+  createLabelDefinition(in: {key: "scenarios", schema: "{\"items\":{\"enum\":[\"DEFAULT\",\"MARKETING\",\"WAREHOUSE\"],\"maxLength\":128,\"pattern\":\"^[A-Za-z0-9]([-_A-Za-z0-9\\\\s]*[A-Za-z0-9])$\",\"type\":\"string\"},\"minItems\":1,\"type\":\"array\",\"uniqueItems\":true}"}) {
+    key
+    schema
+  }
+}
+```
+
+2. Create an Assignment, that Runtimes created by a Warehouse administrator are assigned to the `WAREHOUSE` scenario.
+```graphql
+mutation  {
+  createAutomaticScenarioAssignment(in: {scenarioName: "WAREHOUSE", selector: {key: "owner", value: "warehouse-admin@mycompany.com"}}) {
+    scenarioName
+    selector {
+      key
+      value
+    }
+  }
+}
+```
+4. Register a Runtime, with a label that indicates that it was created by the Warehouse administrator.
+```graphql
+mutation  {
+  registerRuntime(in:{name: "warehouse-runtime-1", labels:{owner:"warehouse-admin@mycompany.com"}}) {
+    name
+    labels
+  }
+}
+```
+
+As you can see from the output below, the Runtime is assigned to the `WAREHOUSE` scenario, thanks to the previously
+defined Automatic Scenario Assignment. 
+```json
+{
+  "data": {
+    "registerRuntime": {
+      "name": "warehouse-runtime-1",
+      "labels": {
+        "owner": "warehouse-admin@mycompany.com",
+        "scenarios": [
+          "WAREHOUSE"
+        ]
+      }
+    }
+  }
+}
+```
+[//]: # TODO: currently runtime is also assigned to the DEFAULT scenario, but should it?
```

Comment:
I don't think it should. It would be great to write it down:
If automatic scenario assignment is specified and new runtime is matched by selector, DEFAULT scenario is not assigned (when this feature is turned on).

Obviously only when we agree that this is how it should work.